app/src/main/java/com/alexbar/layoutbasic/movies_app/screen/detail/widgets/ExpandableSummaryCard.kt

```diff
@@ -0,0 +1,57 @@
+package com.alexbar.layoutbasic.movies_app.screen.detail.widgets
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import com.alexbar.layoutbasic.movies_app.utils.MovieConstants
+import com.alexbar.layoutbasic.ui.theme.AppBackground
+import com.alexbar.layoutbasic.ui.theme.Dimens.dimen_16_dp
+import com.alexbar.layoutbasic.ui.theme.Dimens.dimen_8_dp
+import com.alexbar.layoutbasic.ui.theme.Dimens.max_lines
+import com.alexbar.layoutbasic.ui.theme.Dimens.min_lines
+import com.alexbar.layoutbasic.ui.theme.Typography
+
+@Composable
+fun ExpandableSummaryCard(
+    summary: String,
+    modifier: Modifier = Modifier
+) {
+    var expanded by remember { mutableStateOf(false) }
+
+    Card(
+        colors = CardDefaults.cardColors(
+            containerColor = AppBackground
+        ),
+        shape = RoundedCornerShape(dimen_8_dp),
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(dimen_16_dp)
+            .clickable { expanded = !expanded }
+    ) {
+        Column {
+            Text(
+                text = MovieConstants.summary,
+                style = Typography.bodyMedium
+            )
+            Text(
+                text = summary,
+                maxLines = if (expanded) max_lines else min_lines,
+                style = Typography.labelSmall
+            )
```

Comment:
@ezatpanah I created this expandable card component similarly to the example in the material you shared with me, but in my case, I adjusted the maximum number of lines. I have a question here: how could I programmatically calculate the number of lines the text will occupy? I'd like to know this so that if the description is too short, I won't display the option to expand the component and will only show the option when the text exceeds the default number of lines (4 lines)

Reply:
Regarding your question, I've tested a lot of code. But when it comes to counting lines inside the text view, it seems a bit difficult. All the code I've tried just shows the original lines of text (coming from JSON, and also for example, if you put "\n" inside your text, the code can recognize the text after this as a new line. ) I've tried another way to calculate characters inside the text, but it only shows and hides buttons for showing more or less text with that.

`    val characterCount = summary.length
`
and then

```
            if (characterCount > 200) {
                TextButton(
                    modifier = Modifier.fillMaxWidth(),
                    onClick = {  expanded = !expanded }
                ) {
                    Text(text = if (expanded) MovieConstants.see_less else MovieConstants.see_more)
                }
            }
```

still, I am searching for this one to find a way to use the correct solution for that